src/frameworks/next/index.ts

```diff
@@ -536,6 +536,15 @@ export async function ɵcodegenPublicDirectory(
       let defaultDestPath = isDefaultLocale && join(destDir, basePath, ...destPartsOrIndex);
       if (!fileExistsSync(sourcePath) && fileExistsSync(`${sourcePath}.html`)) {
         sourcePath += ".html";
+        // Check if the HTML file is partial (doesn't have closing </html> tag)
+        const content = await readFile(sourcePath, "utf8");
+        if (!content.includes("</html>")) {
```

Comment:
can we do a similar pass adding these files to `reasonsForBackend`

Reply:
@jamesdaniels I added this and moved some common logic to utils functions and added some tests as well.